### ISSUE TITLE
GLTF Up Axis adjustment is incorrectly applied to CMPT, I3DM formats

### ISSUE THREAD

**Issue by gkjohnson** [NASA-AMMOS/3DTilesRendererJS#184]
The up axis adjustment is not supposed to be applied to PNTS files (#182) and sounds like it needs to be applied to the I3DM GLTF component _before_ the instance transformations and it should be applied individually per CPMT child, which isn't happening right now. Instead of applying the rotation in the load tile function it should be applied _within_ the loaders. This probably means applying a transform to each I3DM instance matrix.

```js
const loader = new B3DMLoader();
loader.adjustmentTransform.makeRotationAxis( X_AXIS, Math.PI / 2 );
loader.load( ... );
```

Test data is needed to verify.

**Comment by RomTreg** [comment]
@gkjohnson Hi, What test data and how much of it do you need?

**Comment by gkjohnson** [comment]
@RomTreg My suspicion is that the GLTF-up adjustment is being applied in the wrong position in the transform hierarchy for both I3DM and CMPT tiles but we aren't using either of those formats ourselves so it's hard to test. For test data we'd want a tileset that contains a CMPT file (that loads an I3DM and / or PNTS file internally) and a tileset that contains an I3DM file that show there's a problem with the transforms similar to how we worked through the PNTS issue. Then we'll be able to dig into the issue (if there is one), find a fix, and then verify that it fixes the display of the test data.

**Comment by Michelle1009** [comment]
I have a tileset file contains CMPT files that loads incorrectly while it is displayed correct in Cesium. I've sent the data to your email as the test data.
![cesium](https://user-images.githubusercontent.com/17328621/174245970-5907591f-2110-4a05-9e14-d676eebf3de8.png)
![threejs](https://user-images.githubusercontent.com/17328621/174245977-408a8bd0-4af2-48db-bcc5-a400d514971f.png)

**Comment by Michelle1009** [comment]
@gkjohnson  I have a tileset file contains CMPT files that loads incorrectly while it is displayed correct in Cesium. I've sent the data to your email as the test data.
![cesium](https://user-images.githubusercontent.com/17328621/174247158-dffbcd1b-955b-4594-8ffd-390e2e67cd0e.png)
![threejs](https://user-images.githubusercontent.com/17328621/174247168-873c4260-ca11-4f99-afa4-8cdab65959b1.png)

**Comment by gkjohnson** [comment]
Thanks @Michelle1009! Would you like to make a PR with the fixes for CMPT and PNTS?

**Comment by Michelle1009** [comment]
I've take some tests but didn't work. Can you give me some help?

test one: I‘ve add the rotation around X axis to I3DMLoader.js, but it didn't make any change.
![image](https://user-images.githubusercontent.com/17328621/174716709-4f9185e4-0f8d-49e0-8add-f22149ccf769.png)

test two: I've changed the instancedMesh.rotation in I3DMLoader.js. the scene changed incorrectly.
![image](https://user-images.githubusercontent.com/17328621/174717056-a286d141-f740-4fe7-af0d-029505326c45.png)
![image](https://user-images.githubusercontent.com/17328621/174717424-2a7e3337-0671-428d-92d6-e1e5aecd4508.png)

**Comment by Michelle1009** [comment]
@gkjohnson  I've take some tests but didn't work. Can you give me some help?

test one: I‘ve add the rotation around X axis to I3DMLoader.js, but it didn't make any change.
![image](https://user-images.githubusercontent.com/17328621/174717745-8883ca2a-0242-451a-b345-27184a3ba6a9.png)


test two: I've changed the instancedMesh.rotation in I3DMLoader.js. the scene changed incorrectly.
![image](https://user-images.githubusercontent.com/17328621/174717767-974f8233-5ae3-4fc0-ae18-9e1f4ebd3017.png)

![image](https://user-images.githubusercontent.com/17328621/174717793-6afe6a5f-97b9-4432-ba7e-6cc551f809a2.png)

**Comment by gkjohnson** [comment]
@Michelle1009 Thanks for taking a look!

Currently the order of matrix operations applied [here](https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/561edea0980baa2b22d903f3db2b8e7cdfc02938/src/three/TilesRenderer.js#L659-L699) is like so:

```
TileTransform * SceneTransform * AxisAdjustment
```

which is correct for glTF. For I3DM loader we need to do the following

```
TileTransform * SceneTransform * InstanceTransform * AxisAdjustment
````

The transform matrix per instance is computed [here](https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/561edea0980baa2b22d903f3db2b8e7cdfc02938/src/three/I3DMLoader.js#L204) - so you should be able to change that line to:

```js
tempMat.compose( tempPos, tempQuat, tempSca ).multiply( axisAdjustmentTransform );
```

You'll also need to make sure you exclude `i3dm` files from having the scene-level adjustment applied [here](https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/561edea0980baa2b22d903f3db2b8e7cdfc02938/src/three/TilesRenderer.js#L687-L691). Feel free to make a "Draft" PR and I can give more specific pointers there!

**Comment by Michelle1009** [comment]
@gkjohnson   I've made a "Draft" PR and the test data loaded correctly after the fix.

**Comment by gkjohnson** [comment]
Fixed in #260
